map.js

```diff
@@ -375,57 +317,58 @@ function initTopbar(){
  *********************************/
 /**
  * Move the map by the specified amount
- * @param {Point} delta delta x and y to move the map
+ * @param {Point} delta delta x and y to move the map, in screen space coords
  */
 function moveMap(delta){
     //increment based on mouse movement
     if(delta){
-        mapOffset.x -= delta.x;
-        mapOffset.y -= delta.y;
+        screenOriginInMapCoords.x -= delta.x;
+        screenOriginInMapCoords.y -= delta.y;
     }
     
-    //clamp values to prevent moving map off screen. //bottom clamp isn't perfect :\
-    if(mapOffset.x < 0) mapOffset.x = 0;
-    if(mapOffset.y < 0) mapOffset.y = 0;
-    //if(mapOffset.x >= img_Map.width - (viewport.clientWidth * zoomLevel)) mapOffset.x = img_Map.width - (viewport.clientWidth * zoomLevel);
-    //if(mapOffset.y >= img_Map.height - viewport.clientHeight) mapOffset.y = img_Map.height - viewport.clientHeight;
-
-    //snap map to top lefthand side if window is too large/map too small.
-    if(img_Map.width < viewport.clientWidth * zoomLevel) mapOffset.x = 0;
-    if(img_Map.height < viewport.clientHeight * zoomLevel) mapOffset.y = 0;
-    
+    //clamp values to prevent moving map off screen.
+    if(screenOriginInMapCoords.x < 0) screenOriginInMapCoords.x = 0;
+    if(screenOriginInMapCoords.y < 0) screenOriginInMapCoords.y = 0;
+
+    const currentMapWidth = img_Map.width / zoomLevel;
+    const currentMapHeight = img_Map.height / zoomLevel;
+    const maxScreenOriginX = Math.max(0,currentMapWidth - viewport.clientWidth);
+    const maxScreenOriginY = Math.max(0,currentMapHeight - viewport.clientHeight);
+    screenOriginInMapCoords.x = Math.min(screenOriginInMapCoords.x, maxScreenOriginX);
+    screenOriginInMapCoords.y = Math.min(screenOriginInMapCoords.y, maxScreenOriginY);
 }
 
 async function initImgs(){
     return new Promise((resolve, reject) =>{
+        var iconsToInit = [
+            "Ayleid",
+            "Camp",
+            "Fort",
+            "Gate",
+            "Cave",
+            "Inn",
+            "Settlement",
+            "Mine",
+            "Landmark",
+            "Shrine",
+            "Nirnroot",
+            "Check",
+            "X"
+        ];
+    
+        iconsToInit.forEach(function(i){
+            icons[i] = document.createElement("IMG");
+            icons[i].width = 48;
+            icons[i].height = 48;
+            icons[i].src = "images/Icon_" + i + ".png";
+            }
+        )
+
```

Comment:
Moved these before the big world map load, so by the time that finishes, all the little icons should be loaded and we can draw correctly.